Tests/UIKitTests/UIRefreshControlExntesionsTests.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  UIRefreshControlExntesionsTests.swift
+//  SwifterSwift
+//
+//  Created by ratul sharker on 7/24/18.
+//  Copyright © 2018 SwifterSwift
+//
+
+#if os(iOS)
+
+import XCTest
+@testable import SwifterSwift
+
+final class UIRefreshControlExtensionTests: XCTestCase {
+
+    func testBeginRefreshAsRefreshControlSubview() {
```

Comment:
I believe that in the [Contributing Guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md#adding-tests) we have a one test function per extension guide covering all the test cases. Also, can you add a test case where you pass send action as `true` to test the behavior on that and maintain the patch diff coverage?

Reply:
I tried to add a function which test the `sendAction` in the following way,

```swift
private var refreshControlValueChangedFired: Bool = false
private var refreshControlValueChangedExpectation: XCTestExpectation!

func testBeginRefreshSendAction() {
    let tableView = UITableView.init()
    let refreshControl = UIRefreshControl()
    refreshControl.addTarget(self, action: #selector(refreshControlValueChanged), for: .valueChanged)
    tableView.addSubview(refreshControl)
    refreshControl.beginRefreshing(in: tableView, animated: true, sendAction: true)
    refreshControlValueChangedExpectation = XCTestExpectation(description: "Refresh Control .valueChanged will fire")
    wait(for: [refreshControlValueChangedExpectation], timeout: 10)
    XCTAssertTrue(refreshControlValueChangedFired)
}

@objc func refreshControlValueChanged() {
    print("this function is called")
    refreshControlValueChangedFired = true
   refreshControlValueChangedExpectation.fulfill()
}
```

But my finding are `refreshControlValueChanged` are never fired. Then i came to know that <a href="https://stackoverflow.com/questions/39851205/how-to-test-if-a-uicontrolevents-has-been-fired">without host application `sendAction` won't work</a>. As it is a library only, no Host application within the SwifterSwift, I believe there is no straight forward way to test the `sendAction` is fired within a library. Please suggest me any other ways to do this.

Reply:
Hey @ratulSharker 
I see the issue here, but actually you don't need to assert if the the sendAction() method triggered the .valueChanged because, is an UIKit API and Apple for sure already tested this. What means you have the garante that if you call that, it will trigger the event, so there's no need to test it. What I mean by adding a test case was just in one of those cases that you are already testing, to put the parameter true just for it to pass there and count as coverage so it increase the path coverage, eventhough you are testing the other behaviors :))
Also can you move the to behaviors tests to only one test method, to follow the [Contributing Guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md#adding-tests) one test method to extension?

Reply:
Thanks @LucianoPAlmeida 
It increases the path coverage.